target/linux/x86/image/Makefile

```diff
@@ -43,6 +44,9 @@ endif
 SIGNATURE:=$(shell perl -e 'printf("%08x", rand(0xFFFFFFFF))')
 ROOTPART:=$(call qstrip,$(CONFIG_TARGET_ROOTFS_PARTNAME))
 ROOTPART:=$(if $(ROOTPART),$(ROOTPART),PARTUUID=$(SIGNATURE)-02)
+EFI_SIGNATURE:=$(strip $(shell uuidgen | sed 's/[a-zA-Z0-9]\{2\}$$/00/'))
```

Comment:
`/usr/bin/uuidgen` is not guaranteed to be present on a Debian system, while it's packaged in `uuid-runtime`, it's not an essential package and probably won't be available in the common buildbot chroot (and neither on any of my full featured development/ office/ multimedia desktops). It might be better to open code this (`/proc/sys/kernel/random/uuid` probably isn't an option either, thinking about non-linux operating systems or containers), to avoid problems like the (temporary) introduction of GNU time as a build dependency.

Reply:
uuidgen is installed in `staging_dir/host/bin` by tools/e2fsprogs, so we do not need `/usr/bin/uuidgen`

Reply:
It doesn't appear to be available early enough though, starting from a fresh clone (nothing prebuilt):
```
$ git clone https://git.openwrt.org/openwrt/openwrt.git/
[...]
$ quilt push -a
Applying patch uefi_0001_firmware-utils-ptgen_add-GPT-support.patch
patching file tools/firmware-utils/Makefile
patching file tools/firmware-utils/src/ptgen.c

Applying patch uefi_0002_grub2_split-to-grub2-and-grub2-efi-packages.patch
patching file package/boot/grub2/Makefile
patching file package/boot/grub2/common.mk
patching file package/boot/grub2/grub2-efi/Makefile
patching file package/boot/grub2/grub2/Makefile

Applying patch uefi_0003_x86_add-EFI-images-and-make-iso-images-EFI-bootable.patch
patching file config/Config-images.in
patching file package/base-files/files/lib/upgrade/common.sh
patching file scripts/gen_image_generic.sh
patching file target/linux/x86/base-files/lib/preinit/79_move_config
patching file target/linux/x86/base-files/lib/upgrade/platform.sh
patching file target/linux/x86/generic/config-4.14
patching file target/linux/x86/generic/config-4.19
patching file target/linux/x86/image/Makefile
patching file target/linux/x86/image/grub-iso.cfg

Now at patch uefi_0003_x86_add-EFI-images-and-make-iso-images-EFI-bootable.patch
$ printf "CONFIG_TARGET_x86=y\nCONFIG_TARGET_x86_64=y\nCONFIG_TARGET_x86_64=y" >.config
$ make defconfig oldconfig
[...]
$ LANG= make 
bash: uuidgen: command not found
time: target/linux/prereq#0.08#0.06#0.14
 make[1] world
 make[2] tools/compile
 make[3] -C tools/flock compile
[...]
```

Reply:
It does no matter,uuidgen only used for build image prereq target does not really need uuidgen

Reply:
@pprindeville Any suggestions here?
Is this ok?
`EFI_SIGNATURE:=$(strip $(shell uuidgen 2>/dev/null | sed 's/[a-zA-Z0-9]\{2\}$$/00/'))`

Reply:
Why are you discarding stderr?  If there's a problem, I'd rather see it... it will appear in the console and not be part of `$EFI_SIGNATURE`, of course.

Reply:
If you use `-r` with `sed` then you can drop the escapes before the curly braces in `\{2\}`.